test/e2e/create_test.go

```diff
@@ -140,38 +140,58 @@ func (tc *testContext) testBYOHConfiguration(t *testing.T) {
 		t.Skip("BYOH testing disabled")
 	}
 
+	err := tc.disableClusterMachineApprover()
+	require.NoError(t, err, "failed to scale down Machine Approver pods")
+
+	err = tc.provisionBYOHConfigMapWithMachineSet()
+	require.NoError(t, err, "error creating BYOH ConfigMap with MachineSets")
+
+	t.Run("VM is configured by ConfigMap controller", func(t *testing.T) {
+		err = tc.waitForWindowsNodes(gc.numberOfBYOHNodes, false, false, true)
+		assert.NoError(t, err, "Windows node creation failed")
+	})
+}
+
+// provisionBYOHConfigMapWithMachineSet provisions BYOH instances using MachineSet and creates the `windows-instances`
+// ConfigMap.
+// The Cluster Machine Approver must be disabled to test BYOH CSR approval feature, so that BYOH instances CSR's are
+// not approved by Cluster Machine Approver
+func (tc *testContext) provisionBYOHConfigMapWithMachineSet() error {
+	_, err := tc.createWindowsMachineSet(gc.numberOfBYOHNodes, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to create Windows MachineSet")
+	}
+	machines, err := tc.waitForWindowsMachines(int(gc.numberOfBYOHNodes), "Provisioned", false)
+	if err != nil {
+		return errors.Wrap(err, "Machines did not reach expected state")
+	}
+	// Change the default shell to PowerShell for the first BYOH VM. We expect there to be BYOH VMs with either
+	// cmd or PowerShell as the default shell, so getting a mix of both, either between different BYOH VMs, or with the
+	// VMs spun up for the Machine testing is important.
+	err = tc.setPowerShellDefaultShell(&machines.Items[0])
+	if err != nil {
+		return errors.Wrapf(err, "unable to change default shell of machine %s", machines.Items[0].GetName())
+	}
+	return tc.createWindowsInstanceConfigMap(machines)
+}
+
+// disableClusterMachineApprover disables the Cluster Machine Approver, by scaling down Cluster Version Operator and
```

Comment:
LGTM though this will probably need to be rebased once https://github.com/openshift/windows-machine-config-operator/pull/873 merges.

Reply:
Correct